### PR TITLE
Feature/improved graph selection

### PR DESCRIPTION
**Pull request by drewbanin**
This works:
```bash
# - run tests for models in ./models/some_dir/**
# - run tests for all snowplow models EXCEPT snowplow_id_map and its children
$ dbt test --models test_project.some.dir.* snowplow.* --exclude 'snowplow_id_map+'
```

### Models
- [x] select models by filesystem hierarchy
- [x] select models by package
- [x] select parents of model
- [x] select children of model
- [x] exclude models by above criteria

### Schema Testing
- [x] source tests from inside of packages
- [x] select tests by package
- [x] select tests by model hierarchy
- [x] select tests by parents of model
- [x] select tests by children of model
- [x] exclude tests by above criteria

### Custom Data Testing
- [x] source tests from inside of packages
- [x] select tests by package
- [x] select tests by model hierarchy
- [x] select tests by parents of model
- [x] select tests by children of model
- [x] exclude tests by above criteria

### Remaining Todo
- [x] Remove `this` package selector
- [x] allow `--models` to be take a space-separated string

### CODE REVIEW

**Comment by drewbanin** [comment]
This branch also fixes https://github.com/analyst-collective/dbt/issues/266

File: dbt/compilation.py [hunk 1]
```diff
@@ -23,6 +23,8 @@
     "models", "data tests", "schema tests", "archives", "analyses"
 ]
 
+GraphFile = 'graph.yml'
```
Comment (cmcarthur) [inline dbt/compilation.py]: nitpicky but why not call this `graph_file_name`?

File: test/unit/test_graph_selection.py [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+        test(('X', 'a', 'b', 'c'), ('X', 'a', '*'), True)
+        test(('X', 'a', 'b', 'c'), ('X', 'a', 'b', '*'), True)
+        test(('X', 'a', 'b', 'c'), ('X', 'a', 'b', 'c'), True)
+        test(('X', 'a', 'b', 'c'), ('X', 'a'), True)
+        test(('X', 'a', 'b', 'c'), ('X', 'a', 'b'), True)
+
+        test(('X', 'a'), ('b'), False)
+        test(('X', 'a'), ('X', 'b'), False)
+        test(('X', 'a'), ('X', 'a', 'b'), False)
+        test(('X', 'a'), ('Y', '*'), False)
```
Comment (cmcarthur) [inline test/unit/test_graph_selection.py]: HOLY SHIT THESE ARE GREAT

File: dbt/compilation.py (hunk 1 above)
Comment (drewbanin) [inline dbt/compilation.py]: :+1:

**Comment by drewbanin** [comment]
@cmcarthur the typo in the `update.sql` file [here](https://github.com/analyst-collective/dbt/blob/development/test/integration/003_simple_reference_test/update.sql#L1) never failed for a couple of reasons:

 - 003_* tests run after the 001_* tests, so the table we were inserting into always existed
 - The 003 test was set up as "run, update, run again". Since the update was a nop, the assertions after the "update" just re-asserted what we had confirmed before the update.
 - We also failed to update the `summary_expected` table to be consistent with the newly inserted data.

Now in the "update", we 1) insert into the correct table and 2) change the value of `summary_expected` to be consistent w/ the newly inserted data [here](https://github.com/analyst-collective/dbt/pull/279/files#diff-c62f5acaf56997199f9a0016bc94ae08).

**Comment by drewbanin** [comment]
@cmcarthur great, going to add those changelog lines then merge later today

**Comment by drewbanin** [comment]
cc @jthandy

**Pull request merged**